### ISSUE TITLE
Locale not supported by C library.

### ISSUE THREAD

**Issue by WhyNotHugo** [sonnyp/Tangram#151]
**Tangram version**

```
➜ flatpak list --app | grep Tangr
Tangram	re.sonny.Tangram	1.3.1	stable	flathub	system
```

**Linux distribution**

ArchLinux`

**Flatpak version**

```
➜ flatpak --version
Flatpak 1.12.2
```

**Desktop environment**

swaywm (tiling compositor, wayland)

**Description**

I run tangram, type any URL, hit enter: nothing happens.

I ran it via a terminal; here's the output:

```
➜ flatpak run re.sonny.Tangram
Gjs-Message: 22:56:50.753: JS LOG: gjs 1.70.0
Gjs-Message: 22:56:50.753: JS LOG: WebKitGTK 2.33.91
Gjs-Message: 22:56:50.753: JS LOG: env: flatpak
Gjs-Message: 22:56:50.753: JS LOG: data_dir: /home/hugo/.var/app/re.sonny.Tangram/data/Tangram
Gjs-Message: 22:56:50.753: JS LOG: cache_dir: /home/hugo/.var/app/re.sonny.Tangram/cache/Tangram
Gjs-Message: 22:56:50.753: JS LOG: config_dir: /home/hugo/.var/app/re.sonny.Tangram/config/Tangram
Gjs-Message: 22:56:50.753: JS LOG: applications_dir: /home/hugo/.local/share/applications
Gjs-Message: 22:56:50.753: JS LOG: keyfile_settings_path:
Gjs-Message: 22:56:50.756: JS LOG: bin: re.sonny.Tangram
Gjs-Message: 22:56:50.756: JS LOG: default_icon: re.sonny.Tangram
Gjs-Message: 22:56:50.756: JS LOG: flag TANGRAM_ENABLE_CUSTOM_APPLICATIONS false
Gjs-Message: 22:56:50.756: JS LOG: flag TANGRAM_ENABLE_CUSTOM_ICONS false
Gjs-Message: 22:56:50.762: JS LOG: programInvocationName: /app/bin/re.sonny.Tangram
Gjs-Message: 22:56:50.762: JS LOG: _: /usr/bin/flatpak
Gjs-Message: 22:56:50.762: JS LOG: pkg.name: re.sonny.Tangram
Gjs-Message: 22:56:50.762: JS LOG: pkg.datadir: /app/share
Gjs-Message: 22:56:50.762: JS LOG: pkg.libdir: /app/lib
Gjs-Message: 22:56:50.762: JS LOG: pkg.localedir: /app/share/locale
Gjs-Message: 22:56:50.762: JS LOG: pkg.moduledir: resource:///re/sonny/Tangram/js
Gjs-Message: 22:56:50.762: JS LOG: pkg.pkglibdir: /app/lib/Tangram
Gjs-Message: 22:56:50.762: JS LOG: pkg.version: 1.3.1
Gjs-Message: 22:56:50.762: JS LOG: pkg.prefix: /app
Gjs-Message: 22:56:50.762: JS LOG: pkg.pkgdatadir: /app/share/Tangram
Gjs-Message: 22:56:50.762: JS LOG: argv /app/bin/re.sonny.Tangram

(re.sonny.Tangram:2): Gtk-WARNING **: 22:56:50.762: Locale not supported by C library.
	Using the fallback 'C' locale.
Gjs-Message: 22:56:50.840: JS LOG: profile.title: Tangram
Gjs-Message: 22:56:50.840: JS LOG: profile.application_id: re.sonny.Tangram
Gjs-Message: 22:56:50.840: JS LOG: profile.settings: /re/sonny/Tangram/
```

Each time I hit enter on the URL bar, I get:

```
flatpak-spawn: Invalid byte sequence in conversion input
Try "flatpak-spawn --help" for more information.
```

Sample input: `https://www.instagram.com/`.

**Comment by jrusnacko** [comment]
My impression is/was that Flatpak apps carry everything they need with them, so the only versions I should care about are the versions of flatpak and the app itself. Is this assumption correct?

I have two machines. On machine 1, this bug does not occur. 
Flatpak version: 1.10.2
Tangram version: 1.4.0

On machine 2, this bug does occur.
Flatpak version: 1.11.2
Tangram version: 1.4.0

I downgraded flatpak on machine 2 to the version 1.10.2, but the bug persists. What am I missing?

**Comment by sonnyp** [comment]
It could be sdks - run `flatpak update` on both.

Could also be hardware related - I imagine both machines are not strictly equal.

**Comment by WhyNotHugo** [comment]
> My impression is/was that Flatpak apps carry everything they need with them, so the only versions I should care about are the versions of flatpak and the app itself. Is this assumption correct?

Tangram is a bit of an exception in that. It tries to run `systemctl`, but that's not installed its package/image. There's two potential fixes for this:

- Check if `systemctl` is executable, and hide the button if it isn't.
- Use systemd's D-Bus API, and configure the flatpak package to have permission for that API.

**Comment by WhyNotHugo** [comment]
Oh, actually, using the D-Bus API won't work because the flatpak can't export a systemd service file: https://github.com/flatpak/flatpak/issues/2787

A transient unit could be used, but that's a bit more complex.

**Comment by jrusnacko** [comment]
I tried `flatpak update` on both, to no avail.

Regarding suggestions by @WhyNotHugo : `systemctl` is executable on both. I am not so knowledgeable to understand the next bit about D-Bus API.

The machines are not the same hardware-wise of course. When I run `flatpak run re.sonny.Tangram` they also show slightly different versions of WebKitGTK and libsoup. Also, on the machine where the bug occurs, it shows this:

`Gtk-Message: 22:05:31.874: Failed to load module "appmenu-gtk-module"`
`Gtk-Message: 22:05:31.926: Failed to load module "canberra-gtk-module"`
`Gtk-Message: 22:05:31.929: Failed to load module "canberra-gtk-module"`

Could that be relevant?

**Comment by sonnyp** [comment]
@WhyNotHugo sorry but I have no idea what you are talking about. How is Tangram related to systemctl? What button ?

@jrusnacko I really don't know - if your objective is to make Tangram work - try updating flatpak to 1.12

**Comment by WhyNotHugo** [comment]
> @WhyNotHugo sorry but I have no idea what you are talking about. How is Tangram related to systemctl? What button ?

Sorry, mixed up threads with something completely unrelated :facepalm:

**Comment by mparker17** [comment]
I'm also experiencing this symptom (i.e.: cannot open any website).

I'm running `Fedora Linux 35.20220322.1 (Silverblue)` on an amd64 (x86_64) system; freshly installed only a few days ago.

I installed Tangram from the Gnome Software program, from the Flathub repo...
```
$ flatpak list | grep Tangram
Name    Application ID          Version Branch  Origin  Installation
Tangram	re.sonny.Tangram	1.4.0	stable	flathub	system

$ flatpak info re.sonny.Tangram

Tangram - Browser for your pinned tabs

          ID: re.sonny.Tangram
         Ref: app/re.sonny.Tangram/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 1.4.0
     License: GPL-3.0+
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 158.7 kB
     Runtime: org.gnome.Platform/x86_64/42
         Sdk: org.gnome.Sdk/x86_64/42

      Commit: 35b1a223b0a36ec47155d878fe0c1ba650ce2d9e3ec351531171e63bc2368b1b
      Parent: d8b20c341e20588a33c86d154d918c48d28c88691fe76297615dda0dd1b5b648
     Subject: Update GNOME runtime to 42 (#18) (5f547083)
        Date: 2022-03-24 19:04:39 +0000
```

When I open the program I see an empty window with an address bar...
![2022-03-26--tangram-initial-state](https://user-images.githubusercontent.com/219351/160257869-d23ae0e1-1ce3-4445-a07d-9a8ad66c9e42.png)
```
$ flatpak run re.sonny.Tangram
Gjs-Message: 17:24:08.480: JS LOG: gjs 1.72.0
Gjs-Message: 17:24:08.480: JS LOG: WebKitGTK 2.36.0
Gjs-Message: 17:24:08.480: JS LOG: libsoup 2.74.2
Gjs-Message: 17:24:08.480: JS LOG: env: flatpak
Gjs-Message: 17:24:08.480: JS LOG: data_dir: /var/home/mparker17/.var/app/re.sonny.Tangram/data/Tangram
Gjs-Message: 17:24:08.480: JS LOG: cache_dir: /var/home/mparker17/.var/app/re.sonny.Tangram/cache/Tangram
Gjs-Message: 17:24:08.480: JS LOG: config_dir: /var/home/mparker17/.var/app/re.sonny.Tangram/config/Tangram
Gjs-Message: 17:24:08.480: JS LOG: applications_dir: /var/home/mparker17/.local/share/applications
Gjs-Message: 17:24:08.480: JS LOG: keyfile_settings_path: 
Gjs-Message: 17:24:08.487: JS LOG: programInvocationName: /app/bin/re.sonny.Tangram
Gjs-Message: 17:24:08.487: JS LOG: _: /usr/bin/flatpak
Gjs-Message: 17:24:08.487: JS LOG: argv /app/bin/re.sonny.Tangram
Failed to open /var/home/mparker17/.var/app/re.sonny.Tangram/data/Tangram/73ec99683b4b4c3f814dd3d46288f93a to expose in sandbox: No such file or directory
```
(note: every time I start the program, the `73ec99683b4b4c3f814dd3d46288f93a` part is different.

When I type an address (with or without the leading `https://`), and press enter, nothing happens, the "Done" button stays greyed-out, and I get the following message written to STDERR/STDOUT (can't tell which)...
```
(process:40): libsoup-WARNING **: 17:24:08.743: Can't open /var/home/mparker17/.var/app/re.sonny.Tangram/data/Tangram/73ec99683b4b4c3f814dd3d46288f93a/cookies.sqlite
Failed to open /var/home/mparker17/.var/app/re.sonny.Tangram/data/Tangram/73ec99683b4b4c3f814dd3d46288f93a to expose in sandbox: No such file or directory
```

If I exit the program, try to add the directory, and start Tangram again, it tries a different directory name, and does not work.
```
$ mkdir -p /var/home/mparker17/.var/app/re.sonny.Tangram/data/Tangram/73ec99683b4b4c3f814dd3d46288f93a

$ flatpak run re.sonny.Tangram
Gjs-Message: 17:35:46.827: JS LOG: gjs 1.72.0
Gjs-Message: 17:35:46.827: JS LOG: WebKitGTK 2.36.0
Gjs-Message: 17:35:46.827: JS LOG: libsoup 2.74.2
Gjs-Message: 17:35:46.827: JS LOG: env: flatpak
Gjs-Message: 17:35:46.827: JS LOG: data_dir: /var/home/mparker17/.var/app/re.sonny.Tangram/data/Tangram
Gjs-Message: 17:35:46.827: JS LOG: cache_dir: /var/home/mparker17/.var/app/re.sonny.Tangram/cache/Tangram
Gjs-Message: 17:35:46.827: JS LOG: config_dir: /var/home/mparker17/.var/app/re.sonny.Tangram/config/Tangram
Gjs-Message: 17:35:46.827: JS LOG: applications_dir: /var/home/mparker17/.local/share/applications
Gjs-Message: 17:35:46.827: JS LOG: keyfile_settings_path: 
Gjs-Message: 17:35:46.833: JS LOG: programInvocationName: /app/bin/re.sonny.Tangram
Gjs-Message: 17:35:46.833: JS LOG: _: /usr/bin/flatpak
Gjs-Message: 17:35:46.834: JS LOG: argv /app/bin/re.sonny.Tangram
Failed to open /var/home/mparker17/.var/app/re.sonny.Tangram/data/Tangram/29f97d60f228445b815f2b95393f435a to expose in sandbox: No such file or directory

(process:40): libsoup-WARNING **: 17:35:47.079: Can't open /var/home/mparker17/.var/app/re.sonny.Tangram/data/Tangram/29f97d60f228445b815f2b95393f435a/cookies.sqlite
Failed to open /var/home/mparker17/.var/app/re.sonny.Tangram/data/Tangram/29f97d60f228445b815f2b95393f435a to expose in sandbox: No such file or directory
```

However, if, in another terminal window, I `mkdir -p /var/home/mparker17/.var/app/re.sonny.Tangram/data/Tangram/29f97d60f228445b815f2b95393f435a`, and then try pressing "Enter" on the URL in the address bar after doing so, it works, and STDERR/STDOUT says...

```
Gjs-Message: 17:41:44.546: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 17:41:44.546: JS LOG: navigation,OTHER,https://google.com/
Gjs-Message: 17:41:44.716: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 17:41:44.717: JS LOG: navigation,OTHER,https://www.google.com/
Gjs-Message: 17:41:44.934: JS LOG: decide-policy,RESPONSE
```

... I can then click "Done" and save the tab.

However, as soon as I create a new tab, and go somewhere else, I get the error again...

```
Failed to open /var/home/mparker17/.var/app/re.sonny.Tangram/data/Tangram/6b4e4c12d1044ce7b5a407b0cc201c1f to expose in sandbox: No such file or directory

(process:106): libsoup-WARNING **: 17:43:10.310: Can't open /var/home/mparker17/.var/app/re.sonny.Tangram/data/Tangram/6b4e4c12d1044ce7b5a407b0cc201c1f/cookies.sqlite
```

*****

Speculating at a fix - maybe Tangram should try to create the directory before trying to use it?

I've been using Tangram for a while now on other computers (Ubuntu Impish amd64, Fedora Silverblue amd64) with no issues... so it is possible that a recent update either to Fedora Silverblue (sometime around 2022-03-05 or later - but it's hard to tell because Silverblue gets daily updates), or Tangram (e.g.: 1.3.2 -> 1.4.0) broke it?

**Comment by ivanqueiroz** [comment]
Im with the same problem here:

```
Failed to open /home/ivanqueiroz/.var/app/re.sonny.Tangram/data/Tangram/7eab50d97e1a4997804c34a518e459b8 to expose in sandbox: Arquivo ou diretório inexistente

(re.sonny.Tangram:2): Gtk-WARNING **: 19:39:32.833: Theme directory actions/16-Dark of theme Zafiro-Icons-Dark-Blue-f has no size field


(process:30): libsoup-WARNING **: 19:39:32.869: Can't open /home/ivanqueiroz/.var/app/re.sonny.Tangram/data/Tangram/7eab50d97e1a4997804c34a518e459b8/cookies.sqlite
```
```
Tangram - Browser for your pinned tabs

        ID: re.sonny.Tangram
       Ref: app/re.sonny.Tangram/x86_64/stable
      Arq.: x86_64
      Ramo: stable
    Versão: 1.4.0
   Licença: GPL-3.0+
    Origem: flathub
   Coleção: org.flathub.Stable
Instalação: user
 Instalado: 158,7 kB
   Runtime: org.gnome.Platform/x86_64/42
       Sdk: org.gnome.Sdk/x86_64/42

    Commit: 35b1a223b0a36ec47155d878fe0c1ba650ce2d9e3ec351531171e63bc2368b1b
       Pai: d8b20c341e20588a33c86d154d918c48d28c88691fe76297615dda0dd1b5b648
   Assunto: Update GNOME runtime to 42 (#18) (5f547083)
      Data: 2022-03-24 19:04:39 +0000
```
I guess the problem is after update Tangram for Gnome 42 (mine is 40.5)

**Comment by tjapro** [comment]
This issue is getting longer without any sign to stop. So, there is some more lines with my report @sonnyp :

The linked issues in this issue are now closed:
- https://github.com/flathub/org.gnome.Epiphany/issues/21#issuecomment-813646993
- https://github.com/flatpak/flatpak/issues/3712

Also my system is fully update, right before the Fedora 36 release (the Fedora 36 Beta release was March 28, 2022 = https://fedoramagazine.org/announcing-fedora-36-beta/):

```
OS	- Fedora Linux 35 (Workstation Edition) x86_64 
DE	- GNOME 41.4 
WM	- Mutter  
Kernel	- 5.16.17-200.fc35.x86_64 
```

The flatpak program info:

```
Tangram - Browser for your pinned tabs

          ID: re.sonny.Tangram
         Ref: app/re.sonny.Tangram/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 1.4.0
     License: GPL-3.0+
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 158,7 kB
     Runtime: org.gnome.Platform/x86_64/42
         Sdk: org.gnome.Sdk/x86_64/42

      Commit: 35b1a223b0a36ec47155d878fe0c1ba650ce2d9e3ec351531171e63bc2368b1b
      Parent: d8b20c341e20588a33c86d154d918c48d28c88691fe76297615dda0dd1b5b648
     Subject: Update GNOME runtime to 42 (#18) (5f547083)
        Date: 2022-03-24 19:04:39 +0000
```

The flatpak logs:
```
flatpak run re.sonny.Tangram                                    
Gjs-Message: 00:19:27.593: JS LOG: gjs 1.72.0
Gjs-Message: 00:19:27.593: JS LOG: WebKitGTK 2.36.0
Gjs-Message: 00:19:27.593: JS LOG: libsoup 2.74.2
Gjs-Message: 00:19:27.593: JS LOG: env: flatpak
Gjs-Message: 00:19:27.593: JS LOG: data_dir: /home/..../.var/app/re.sonny.Tangram/data/Tangram
Gjs-Message: 00:19:27.593: JS LOG: cache_dir: /home/..../.var/app/re.sonny.Tangram/cache/Tangram
Gjs-Message: 00:19:27.593: JS LOG: config_dir: /home/..../.var/app/re.sonny.Tangram/config/Tangram
Gjs-Message: 00:19:27.593: JS LOG: applications_dir: /home/..../.local/share/applications
Gjs-Message: 00:19:27.593: JS LOG: keyfile_settings_path: 
Gjs-Message: 00:19:27.610: JS LOG: programInvocationName: /app/bin/re.sonny.Tangram
Gjs-Message: 00:19:27.610: JS LOG: _: null
Gjs-Message: 00:19:27.611: JS LOG: argv /app/bin/re.sonny.Tangram
Gtk-Message: 00:19:27.652: Failed to load module "canberra-gtk-module"
Gtk-Message: 00:19:27.652: Failed to load module "pk-gtk-module"
Gtk-Message: 00:19:27.656: Failed to load module "canberra-gtk-module"
Gtk-Message: 00:19:27.656: Failed to load module "pk-gtk-module"
Gtk-Message: 00:19:28.734: Failed to load module "canberra-gtk-module"
Gtk-Message: 00:19:28.734: Failed to load module "pk-gtk-module"
Gtk-Message: 00:19:28.736: Failed to load module "canberra-gtk-module"
Gtk-Message: 00:19:28.736: Failed to load module "pk-gtk-module"
Gtk-Message: 00:19:28.788: Failed to load module "canberra-gtk-module"
Gtk-Message: 00:19:28.789: Failed to load module "pk-gtk-module"
Gtk-Message: 00:19:28.790: Failed to load module "canberra-gtk-module"
Gtk-Message: 00:19:28.791: Failed to load module "pk-gtk-module"
Gjs-Message: 00:19:32.991: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 00:19:32.991: JS LOG: navigation,OTHER,https://web.whatsapp.com/
Gjs-Message: 00:19:32.992: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 00:19:32.992: JS LOG: navigation,OTHER,https://web.whatsapp.com/
Gtk-Message: 00:19:33.428: Failed to load module "canberra-gtk-module"
Gtk-Message: 00:19:33.428: Failed to load module "pk-gtk-module"
Gtk-Message: 00:19:33.430: Failed to load module "canberra-gtk-module"
Gtk-Message: 00:19:33.431: Failed to load module "pk-gtk-module"
Gjs-Message: 00:19:33.769: JS LOG: decide-policy,RESPONSE
Gjs-Message: 00:19:33.771: JS LOG: decide-policy,RESPONSE
Failed to open /home/..../.var/app/re.sonny.Tangram/data/Tangram/fe09add90bb04f4793531aa615c7ec9a to expose in sandbox: Ficheiro ou pasta inexistente

(process:12079): libsoup-WARNING **: 00:20:09.735: Can't open /home/..../.var/app/re.sonny.Tangram/data/Tangram/fe09add90bb04f4793531aa615c7ec9a/cookies.sqlite
Failed to open /home/.../.var/app/re.sonny.Tangram/data/Tangram/fe09add90bb04f4793531aa615c7ec9a to expose in sandbox: Ficheiro ou pasta inexistente

(re.sonny.Tangram:2): Gjs-CRITICAL **: 00:20:51.709: JS ERROR: Gio.IOErrorEnum: Failed to open /home/..../.var/app/re.sonny.Tangram/data/Tangram/fe09add90bb04f4793531aa615c7ec9a
destroy@file:///app/share/Tangram/instances.js:90:40
onCancelNewTab@file:///app/share/Tangram/window.js:220:20
main@file:///app/share/Tangram/main.js:59:22
@file:///app/bin/re.sonny.Tangram:18:5
```

I do not tried the workaround, because it will disable sandboxing and 

> browser tabs will no longer be subsandboxed separately from each other

which is a not good for me: it will break the separation of whatsapp accounts.

Also, as this comment says https://github.com/flathub/org.gnome.Epiphany/issues/21#issuecomment-814079653 :
> Since there are no changes to make in Epiphany, I'm going to close this in favor of any of [flatpak/flatpak#3514](https://github.com/flatpak/flatpak/issues/3514) [flatpak/flatpak#3563](https://github.com/flatpak/flatpak/issues/3563) [flatpak/flatpak#3712](https://github.com/flatpak/flatpak/issues/3712), which are all duplicates of each other. Once those are fixed, Epiphany will stop crashing. In the meantime, workaround is to manually disable the sandbox or not use mixed locales.

And those issues are closed, and one of them are just for tracking.

In conclusion, I'll be happy to see some feedback from the owner of this project @sonnyp (or from other collaborators) with any solutions, blockers, or with an update that fixes this issue.

Thanks in advance!

**Comment by esijg** [comment]
Seem to be getting the same issue on a near clean install of Fedora 35. Can't load any website at all in Tangram. Just gives me a blank window.

```
[esi@melkorka ~]$ flatpak --version
Flatpak 1.12.7
[esi@melkorka ~]$ flatpak list | grep Tangram
Tangram	re.sonny.Tangram	1.4.0	stable	flathub	system
[esi@melkorka ~]$ flatpak info re.sonny.Tangram

Tangram - Browser for your pinned tabs

          ID: re.sonny.Tangram
         Ref: app/re.sonny.Tangram/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 1.4.0
     License: GPL-3.0+
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 158,7 kB
     Runtime: org.gnome.Platform/x86_64/42
         Sdk: org.gnome.Sdk/x86_64/42

      Commit: 35b1a223b0a36ec47155d878fe0c1ba650ce2d9e3ec351531171e63bc2368b1b
      Parent: d8b20c341e20588a33c86d154d918c48d28c88691fe76297615dda0dd1b5b648
     Subject: Update GNOME runtime to 42 (#18) (5f547083)
        Date: 2022-03-24 19:04:39 +0000
```

Terminal output

```
[esi@melkorka ~]$ flatpak run re.sonny.Tangram
Gjs-Message: 14:11:43.456: JS LOG: gjs 1.72.0
Gjs-Message: 14:11:43.456: JS LOG: WebKitGTK 2.36.0
Gjs-Message: 14:11:43.456: JS LOG: libsoup 2.74.2
Gjs-Message: 14:11:43.457: JS LOG: env: flatpak
Gjs-Message: 14:11:43.457: JS LOG: data_dir: /home/esi/.var/app/re.sonny.Tangram/data/Tangram
Gjs-Message: 14:11:43.457: JS LOG: cache_dir: /home/esi/.var/app/re.sonny.Tangram/cache/Tangram
Gjs-Message: 14:11:43.457: JS LOG: config_dir: /home/esi/.var/app/re.sonny.Tangram/config/Tangram
Gjs-Message: 14:11:43.457: JS LOG: applications_dir: /home/esi/.local/share/applications
Gjs-Message: 14:11:43.457: JS LOG: keyfile_settings_path: 
Gjs-Message: 14:11:43.474: JS LOG: programInvocationName: /app/bin/re.sonny.Tangram
Gjs-Message: 14:11:43.474: JS LOG: _: /usr/bin/flatpak
Gjs-Message: 14:11:43.476: JS LOG: argv /app/bin/re.sonny.Tangram
Failed to open /home/esi/.var/app/re.sonny.Tangram/data/Tangram/1db8f2f222dd4c67a11ea9a994158cb2 to expose in sandbox: No such file or directory

(process:37): libsoup-WARNING **: 14:11:44.812: Can't open /home/esi/.var/app/re.sonny.Tangram/data/Tangram/1db8f2f222dd4c67a11ea9a994158cb2/cookies.sqlite
Failed to open /home/esi/.var/app/re.sonny.Tangram/data/Tangram/1db8f2f222dd4c67a11ea9a994158cb2 to expose in sandbox: No such file or directory
Failed to open /home/esi/.var/app/re.sonny.Tangram/data/Tangram/1db8f2f222dd4c67a11ea9a994158cb2 to expose in sandbox: No such file or directory
```

The "failed to open" appears every time i try to load a website. In this case, instagram.com. I've also tested multiple other websites with same result. Epiphany on the other hand seems to work for me.

**Comment by philn** [comment]
> Ok a potential workaround for now is
> 
> ```
> WEBKIT_FORCE_SANDBOX=0 flatpak run re.sonny.Tangram
> ```
> 
> [flathub/org.gnome.Epiphany#21 (comment)](https://github.com/flathub/org.gnome.Epiphany/issues/21#issuecomment-813646993)
> 
> > (You're still protected by flatpak's primary sandbox, but browser tabs will no longer be subsandboxed separately from each other.

Should this env var be set in the flatpak manifest? Otherwise Tangram is pretty much unusable :(

**Comment by davidcbenson** [comment]
I also have this issue, with essentially identical. I am running Ubuntu, though. I don't know if that matters. 

I have tried running Epiphany, and it runs without problems. I didn't see that pointed out in the issues above.

**Comment by rcchan123** [comment]
Same with me on fedora 35:

On startup
```
[ralphchan@ralph-laptop ~]$ flatpak run re.sonny.Tangram
Gjs-Message: 01:08:33.987: JS LOG: gjs 1.72.0
Gjs-Message: 01:08:33.987: JS LOG: WebKitGTK 2.36.0
Gjs-Message: 01:08:33.987: JS LOG: libsoup 2.74.2
Gjs-Message: 01:08:33.988: JS LOG: env: flatpak
Gjs-Message: 01:08:33.988: JS LOG: data_dir: /home/ralphchan/.var/app/re.sonny.Tangram/data/Tangram
Gjs-Message: 01:08:33.988: JS LOG: cache_dir: /home/ralphchan/.var/app/re.sonny.Tangram/cache/Tangram
Gjs-Message: 01:08:33.988: JS LOG: config_dir: /home/ralphchan/.var/app/re.sonny.Tangram/config/Tangram
Gjs-Message: 01:08:33.988: JS LOG: applications_dir: /home/ralphchan/.local/share/applications
Gjs-Message: 01:08:33.988: JS LOG: keyfile_settings_path: 
Gjs-Message: 01:08:33.996: JS LOG: programInvocationName: /app/bin/re.sonny.Tangram
Gjs-Message: 01:08:33.996: JS LOG: _: /usr/bin/flatpak
Gjs-Message: 01:08:33.997: JS LOG: argv /app/bin/re.sonny.Tangram
Failed to open /home/ralphchan/.var/app/re.sonny.Tangram/data/Tangram/01b683eee37f418fa1ba2f126a1f0013 to expose in sandbox: No such file or directory

(process:27): libsoup-WARNING **: 01:08:34.369: Can't open /home/ralphchan/.var/app/re.sonny.Tangram/data/Tangram/01b683eee37f418fa1ba2f126a1f0013/cookies.sqlite
```
When I try to connect to services:
`
Failed to open /home/ralphchan/.var/app/re.sonny.Tangram/data/Tangram/01b683eee37f418fa1ba2f126a1f0013 to expose in sandbox: No such file or directory`
                 
I also tried the workaround shown here, but it only works occationally

> Ok a potential workaround for now is
> 
> ```
> WEBKIT_FORCE_SANDBOX=0 flatpak run re.sonny.Tangram
> ```
> 
> [flathub/org.gnome.Epiphany#21 (comment)](https://github.com/flathub/org.gnome.Epiphany/issues/21#issuecomment-813646993)
> 
> > (You're still protected by flatpak's primary sandbox, but browser tabs will no longer be subsandboxed separately from each other.

Terminal output on startup:

```
[ralphchan@ralph-laptop ~]$ WEBKIT_FORCE_SANDBOX=0 flatpak run re.sonny.Tangram
Gjs-Message: 01:13:13.343: JS LOG: gjs 1.72.0
Gjs-Message: 01:13:13.343: JS LOG: WebKitGTK 2.36.0
Gjs-Message: 01:13:13.343: JS LOG: libsoup 2.74.2
Gjs-Message: 01:13:13.344: JS LOG: env: flatpak
Gjs-Message: 01:13:13.344: JS LOG: data_dir: /home/ralphchan/.var/app/re.sonny.Tangram/data/Tangram
Gjs-Message: 01:13:13.344: JS LOG: cache_dir: /home/ralphchan/.var/app/re.sonny.Tangram/cache/Tangram
Gjs-Message: 01:13:13.344: JS LOG: config_dir: /home/ralphchan/.var/app/re.sonny.Tangram/config/Tangram
Gjs-Message: 01:13:13.344: JS LOG: applications_dir: /home/ralphchan/.local/share/applications
Gjs-Message: 01:13:13.344: JS LOG: keyfile_settings_path: 
Gjs-Message: 01:13:13.353: JS LOG: programInvocationName: /app/bin/re.sonny.Tangram
Gjs-Message: 01:13:13.353: JS LOG: _: /usr/bin/flatpak
Gjs-Message: 01:13:13.354: JS LOG: argv /app/bin/re.sonny.Tangram

(process:27): libsoup-WARNING **: 01:13:13.658: Can't open /home/ralphchan/.var/app/re.sonny.Tangram/data/Tangram/7d37412a6b0c441ba01fbc63bdfe59fe/cookies.sqlite
```

And here is when I tried multiple times to open gmail

```
(process:118): libsoup-WARNING **: 01:15:14.316: Can't open /home/ralphchan/.var/app/re.sonny.Tangram/data/Tangram/450c5d2bad484d8ea22d5d321b262627/cookies.sqlite
Gjs-Message: 01:15:14.328: JS LOG: decide-policy,RESPONSE
Gjs-Message: 01:15:17.357: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 01:15:17.357: JS LOG: navigation,OTHER,https://accounts.youtube.com/accounts/CheckConnection?pmpo=https%3A%2F%2Faccounts.google.com&v=-510136939&timestamp=1649870117354
Gjs-Message: 01:15:33.506: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 01:15:33.506: JS LOG: navigation,OTHER,https://accounts.google.com/_/bscframe
Gjs-Message: 01:15:33.827: JS LOG: decide-policy,RESPONSE
Gjs-Message: 01:15:33.896: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 01:15:33.896: JS LOG: navigation,OTHER,about:blank
Gjs-Message: 01:15:37.153: JS LOG: decide-policy,RESPONSE

(re.sonny.Tangram:2): Gjs-CRITICAL **: 01:15:47.800: JS ERROR: Gio.IOErrorEnum: Trash portal failed on /home/ralphchan/.var/app/re.sonny.Tangram/data/Tangram/450c5d2bad484d8ea22d5d321b262627
destroy@file:///app/share/Tangram/instances.js:90:40
onCancelNewTab@file:///app/share/Tangram/window.js:220:20
main@file:///app/share/Tangram/main.js:59:22
@file:///app/bin/re.sonny.Tangram:18:5
```
However, when I keep trying, it eventually works.

**Comment by sonnyp** [comment]
This issue was about the following errors

>(re.sonny.Tangram:2): Gtk-WARNING **: 13:14:44.391: Locale not supported by C library.
	Using the fallback 'C' locale.

and

> flatpak-spawn: Invalid byte sequence in conversion input

which have been fixed by flatpak

See

* https://github.com/flathub/org.gnome.Epiphany/issues/21
* https://github.com/flatpak/flatpak/issues/3712

so I'm gonna close it.

---

Regarding the issue

> Failed to open ... to expose in sandbox: No such file or directory

please file a new issue

https://github.com/sonnyp/Tangram/issues/new?assignees=&labels=&template=bug_report.md&title=

so we can start with a clean room
